### PR TITLE
Added external worker documentation

### PR DESCRIPTION
**Pull request by droyad**

### CODE REVIEW

File: docs/administration/workers/index_[2018.2,).md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: Workers
+description: Worker setup and configuration.
+position: 1400
+---
+
+When your deployment process runs a script or extracts a package without involving a Tentacle or SSH target, they process is performed by a worker (see [where steps run](#where-steps-run) below). By default the worker lives inside the Octopus Server process (the built-in worker), however you can also configure an [external worker](#external-worker). 
+
+In many cases running user provided scripts under the context of the Octopus Server is undesirable, so we have
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: Incomplete sentence here :)

File: docs/administration/workers/index_[2018.2,).md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+title: Workers
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: Really good to see this getting its own section now in administration!

File: docs/administration/workers/index_[2018.2,).md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+When your deployment process runs a script or extracts a package without involving a Tentacle or SSH target, they process is performed by a worker (see [where steps run](#where-steps-run) below). By default the worker lives inside the Octopus Server process (the built-in worker), however you can also configure an [external worker](#external-worker). 
+
+In many cases running user provided scripts under the context of the Octopus Server is undesirable, so we have 
+
+## Built-in Worker
+
+Any time a script needs to be run (e.g. PowerShell or Calamari), Octopus Server would spawned a new process from it's process it's security context. In `2018.1.0` we introduced the ability to [change the security context](docs/administration/security/built-in-worker) under which those new processes spawned, adding a layer of security.
+
+:::hint
+Future versions of Octopus Server will bundle a worker with the server that is started automatically. It will be run under the user contect supplied through the `builtin-worker` command.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: This sentence has some typos, and I'm not sure of its purpose. I'm actually not sure if there's much point opening up the "black box" here. I wonder if the customers care about how we "run things in a different security context"?

File: docs/administration/workers/index_[2018.2,).md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+position: 1400
+---
+
+When your deployment process runs a script or extracts a package without involving a Tentacle or SSH target, they process is performed by a worker (see [where steps run](#where-steps-run) below). By default the worker lives inside the Octopus Server process (the built-in worker), however you can also configure an [external worker](#external-worker). 
+
+In many cases running user provided scripts under the context of the Octopus Server is undesirable, so we have 
+
+## Built-in Worker
+
+Any time a script needs to be run (e.g. PowerShell or Calamari), Octopus Server would spawned a new process from it's process it's security context. In `2018.1.0` we introduced the ability to [change the security context](docs/administration/security/built-in-worker) under which those new processes spawned, adding a layer of security.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: I like the wording you've used here about "change the security context". :+1:

File: docs/administration/workers/index_[2018.2,).md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+:::hint
+Future versions of Octopus Server will bundle a worker with the server that is started automatically. It will be run under the user contect supplied through the `builtin-worker` command.
+:::
+
+## External Worker
+
+In `2018.2.0` we added the ability to nominate a Tentacle to perform the role of the worker. This Tentacle can run under a different user account and also on a different machine, adding extra layers of security. The built in worker is automatically disabled when an external worker is configured, preventing the Octopus Server from running any user provided scripts.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: I've tended to use `built-in` instead of `built in`

Comment (michaelnoonan) [reply docs/administration/workers/index_[2018.2,).md]: Perhaps: "preventing any user-provided scripts from being executed by the Octopus Server itself."?

File: docs/administration/workers/index_[2018.2,).md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+:::hint
+Future versions of Octopus Server will expand on the worker concept, allowing you to create worker pools with multiple workers. Steps then can be configured to run on those pools.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: Perhaps it's worth adding a link to the spec here which we'll remove when shipped?

File: docs/administration/workers/index_[2018.2,).md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+When that feature is released, the commandline options will be removed. If a external worker is configured, it will automatically be added to the default worker pool.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: "If **an** external worker is configured"

File: docs/administration/workers/index_[2018.2,).md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+:::
+
+### Setup
+
+To enable the external worker, you will need to setup a Tentacle, either on the same machine as the server or a different machine. You can use an existing Tentacle that is used as a deployment target. We recommend that the Tentacle is on the same local network as the server to speed up package transfers.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: "**set up** a Tentacle"

File: docs/administration/workers/index_[2018.2,).md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+Once you have done that, and have it's address and thumbprint, run the following command.
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: **it's** should be **its**

File: docs/administration/workers/index_[2018.2,).md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+After restarting, the built-in worker will be disabled. If you want to revert to the built-in worker, run the following command.
+
+```
+Octopus.Server external-worker --reset
+```
+
+### Limitations
+The external worker currently has the following limitations
+- There can only be one
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: Insert highlander reference ;)

File: docs/administration/workers/index_[2018.2,).md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+- It does not participate in Health Checks unless it is also a deployment target
+- Packages are always transfered from the server to the worker, download on target is not supported
+- The worker will run steps from different projects simultaneously (keeping the behaviour of the built-in worker), which could allow one project to access the working folder of another project 
+
+## Where steps run
```
Comment (michaelnoonan) [inline docs/administration/workers/index_[2018.2,).md]: Needs an anchor for the link at the top of the page to work

File: docs/api-and-integration/octopus.server.exe-command-line/builtin-worker_[2018.1,).md [hunk 12]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (michaelnoonan) [inline docs/api-and-integration/octopus.server.exe-command-line/builtin-worker_[2018.1,).md]: Whoops, I completely forgot to add this page! Thanks!

File: docs/api-and-integration/octopus.server.exe-command-line/builtin-worker_[2018.1,).md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+title: Octopus Server commands - Built-in Worker
+description: Creates a new certificate that Octopus server can use to authenticate itself with its Tentacles
```
Comment (michaelnoonan) [inline docs/api-and-integration/octopus.server.exe-command-line/builtin-worker_[2018.1,).md]: Wrong description? Copy-pasta

File: docs/api-and-integration/octopus.server.exe-command-line/external-worker_[2018.2,).md [hunk 14]
```diff
@@ -0,0 +1,28 @@
+---
+title: Octopus Server commands - External Worker
+description: Creates a new certificate that Octopus server can use to authenticate itself with its Tentacles
```
Comment (michaelnoonan) [inline docs/api-and-integration/octopus.server.exe-command-line/external-worker_[2018.2,).md]: Wrong description?

File: docs/administration/workers/index_[2018.2,).md (hunk 3 above)
Comment (droyad) [reply docs/administration/workers/index_[2018.2,).md]: Good point

File: docs/administration/workers/index_[2018.2,).md (hunk 11 above)
Comment (droyad) [reply docs/administration/workers/index_[2018.2,).md]: I'm pretty sure they are auto-generated, but I'll check once it goes live

**Pull request merged**